app/src/main/kotlin/com/wire/android/ui/login/Login.kt

```diff
@@ -0,0 +1,204 @@
+        keyboardOptions = KeyboardOptions(
+            keyboardType = KeyboardType.Password,
+            imeAction = ImeAction.Done
+        ),
+        keyboardActions = KeyboardActions(onDone = { keyboardController?.hide() }),
+        visualTransformation = if (passwordVisibility) VisualTransformation.None else PasswordVisualTransformation(),
+        trailingIcon = {
+            val image = if (passwordVisibility) Icons.Filled.Visibility else Icons.Filled.VisibilityOff
+            IconButton(onClick = { passwordVisibility = !passwordVisibility }) {
+                Icon(imageVector = image, "", modifier = Modifier.size(20.dp))
+            }
+        },
+        modifier = modifier,
+    )
+}
+
+@Composable
+private fun ForgotPasswordLabel(modifier: Modifier) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = modifier) {
+        val context = LocalContext.current
+        Text(
+            text = stringResource(R.string.login_forgot_password),
+            style = MaterialTheme.typography.body1.copy(
+                textDecoration = TextDecoration.Underline,
+                color = MaterialTheme.colors.primary
+            ),
+            textAlign = TextAlign.Center,
+            modifier = Modifier
+                .clickable(
+                    interactionSource = remember { MutableInteractionSource() },
+                    indication = null,
+                    onClick = { Toast.makeText(context, "Forgot password click 💥", Toast.LENGTH_SHORT).show() } //TODO
+                )
+        )
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun LoginButton(modifier: Modifier, email: String, password: String) {
+    var isLoading by remember { mutableStateOf(false) }
+    val interactionSource = remember { MutableInteractionSource() }
+    Column(modifier = modifier) {
+        val enabled = validInput(email, password) && !isLoading
+        Button(
+            interactionSource = interactionSource,
+            shape = RoundedCornerShape(16.dp),
+            elevation = ButtonDefaults.elevation(0.dp, 0.dp, 0.dp),
+            colors = AnimatedButtonColors(enabled = enabled),
+            onClick = { isLoading = true }, //TODO
+            enabled = validInput(email, password) && !isLoading,
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(48.dp)
+        ) {
+            Box(modifier = Modifier.fillMaxWidth()) {
+                val text = if(isLoading) stringResource(R.string.label_logging_in) else stringResource(R.string.label_login)
+                Text(
+                    text = text,
+                    textAlign = TextAlign.Center,
+                    style = MaterialTheme.typography.button,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = 32.dp),
+                )
+                androidx.compose.animation.AnimatedVisibility(
```

Comment:
why this import here ?

Reply:
There is a bug, probably in the language, take a look here: https://stackoverflow.com/questions/67975569/why-cant-i-use-animatedvisibility-in-a-boxscope

The workaround is to use the fully qualified name 🤷